### ISSUE TITLE
Failed to connect: Invalid Connect Token

### ISSUE THREAD

**Issue by Danila24ru** [KillaMaaki/Netcode.IO.NET#2]
`logLine("Generating token on client...");
        TokenFactory factory = new TokenFactory(ProtocolID, privateKey);
        byte[] connectToken = factory.GenerateConnectToken(ipEndPoint, 30, 1UL, 1UL, new byte[256]);

        client.Connect(connectToken, () =>
		   { ...
`
So, its bring this message in unity log:

> State changed to: -5
UnityEngine.Debug:Log(Object)
UnityNetcodeIO.<Connect>c__AnonStorey0:<>m__0(ClientState) (at Assets/UnityNetcodeIO/Scripts/Core/NetcodeClient.cs:342)
NetcodeIO.NET.Client:changeState(ClientState)
NetcodeIO.NET.Client:Connect(Byte[])
UnityNetcodeIO.NetcodeClient:Connect(Byte[], Action, Action`1) (at Assets/UnityNetcodeIO/Scripts/Core/NetcodeClient.cs:361)
<connectToServer>c__Iterator0:MoveNext() (at Assets/UnityNetcodeIO/Test/Scripts/TestNetcodeIOClient.cs:84)
UnityEngine.SetupCoroutine:InvokeMoveNext(IEnumerator, IntPtr)

**Comment by GlaireDaggers** [comment]
Oh wow. I'm reasonably sure I had this fixed before. There was a point where I screwed up a git commit and several files seemed to have been reverted at random. I didn't look through TokenFactory well enough apparently. There were at least three or four separate areas where connect tokens were not being serialized properly (missing version info string, expire timestamp was zero, etc).
I've stepped through and fixed the issues causing an invalid token and pushed up a commit.
